### DISCUSSION TITLE
Added a model but can’t update it, and friends.

### DISCUSSION THREAD

**Discussion by atljoseph** [plandex-ai/plandex#146] [Ideas]
1.
Can’t update model. Have to delete and readd. This is tough because the inputs aren’t all clear and it needs to be easier to recover from inputting the wrong thing. 

Examples where I made mistake:
- API key env var name. I thought it was asking for api key. 
- Convo token limit update. This feels a bit … cryptic. It’d be helpful if the cli outputted a visual example of how all the token limits play together. 

2.
After deleting a model it is not removed from the any plans. There should be a way for the user to declare the successor to the model they are removing, or even set all references of it to use a default model. Even if it just outright refused to do the removal until the model was removed from all models sets, that would make more sense. 

3. 
After re-adding a new model with same name as before, the plan still uses settings from previously deleted model. Have to go back into the set model flow and reset everything from scratch. 

4.
Set model flow exits every time you edit one record. Needs to stay open. We invent software to help us automate, and not make more work. 

5. 
Is confusing between set model and set model default. Set model default sounds like it should update models to default, but it is not that. Maybe that should be called set model *org*, since it sets the models for the *org*. 

6.
Am a bit confused by the changes view. Diff makes 100% sense, but the other view… sometimes it shows stuff that didn’t change, or… idk what that thing does LOL. Seems like it has a purpose to relate the LLMs reasoning for doing certain things, but I didn’t walk away with a clear understanding, even after using it many times… but the UI is cool.

**Comment by danenania** [comment]
Thanks @atljoseph this is good feedback. I agree the model management workflow can be made smoother.

The `plandex changes` TUI is just meant to show the changes accumulated for each file. Could you paste a screenshot perhaps of an example where you’re confused by what it’s showing?
